app/lib/feature_flags.rb

```diff
@@ -22,19 +27,24 @@ def enabled?(flag, session_data = nil, without_session_data: false)
         raise "Pass in session_data or set without_session_data to true"
       end
 
+      #  if the flag exists in the session_data use that setting
       if session_data && session_data["feature_flags"]&.key?(flag.to_s)
         return session_data["feature_flags"][flag.to_s]
       end
 
-      if overrideable?
-        override = FeatureFlagOverride.find_by(key: flag)
-        if override
-          return override.value
-        end
-      end
+      possible_override = overrideable_flag(flag)
+      return possible_override unless possible_override.nil?
 
       if STATIC_FLAGS.key?(flag)
-        ENV["#{flag.to_s.upcase}_FEATURE_FLAG"]&.casecmp("enabled")&.zero? || STATIC_FLAGS.fetch(flag).fetch(:default)
+        # if the flag is a valid flag and of type `session` but does not
+        # exist in the session_data we return the default value for that flag
+        if STATIC_FLAGS[flag][:type] == "session"
+          STATIC_FLAGS[flag][:default]
+        else
+          # if it is a global flag we check the env_var
+          # if there is no env_var specified we use the default value for that flag
```

Comment:
I'm not sure if this is a bit scary or not. Why would we use the default value if there is no env_var? Could we risk accidentally turning something on?

Reply:
In fact maybe I am misunderstanding here. Is this setting purely for testing? Where we might want to have a default but there is no env var? I guess my fear is accidentally leaving a `default: true` and forgetting to set an env_var in prod. Maybe I am overthinking it. Maybe a comment would help here.

Reply:
Yes this is only ever expected to fire during testing when we provide an env_var in the before block and no session_data.
The defaults are also for tests only, they are added so that when a flag doesn't have a value we give it a value that will not break our tests.
I will add more to the comment to explain the env_var in tests.

Reply:
i added a description as to what the default does rather than rename the attribute